exams/utils.py

```diff
@@ -59,7 +68,105 @@ def authorize_for_exam(mmtrack, course_run):
                 date_last_eligible=add_year(now)
             )
             log.info(
-                'user "%s" is authorize for exam the for course id "%s"',
+                '[Exam authorization] user "%s" is authorize for exam the for course id "%s"',
                 mmtrack.user.username,
                 course_run.edx_course_key
             )
+        else:
+            errors_message = message_not_passed_or_exist_template.format(
+                user=mmtrack.user.username,
+                course_id=course_run.edx_course_key
+            )
+            raise ExamAuthorizationException(errors_message)
+    else:
+        errors_message = message_not_eligible_template.format(
+            user=mmtrack.user.username,
+            course_id=course_run.edx_course_key
+        )
+        raise ExamAuthorizationException(errors_message)
```

Comment:
please refactor this code in the following way (please verify that it actually works):
```
def authorize_for_exam(mmtrack, course_run):
    """
    Authorize user for exam if he has paid for course and passed course.
    Args:
        mmtrack (dashboard.utils.MMTrack): a instance of all user information about a program.
        course_run (CourseRun): A CourseRun object.
    """
    if not mmtrack.course_has_exam(course_run) and mmtrack.has_paid(course_run.edx_course_key):
        raise ExamAuthorizationException(
            message_not_eligible_template.format(
                user=mmtrack.user.username,
                course_id=course_run.edx_course_key
            )
        )

    now = datetime.datetime.now(tz=pytz.UTC)
    # if user paid for a course then create his exam profile if it is not creaated yet.
    ExamProfile.objects.get_or_create(profile=mmtrack.user.profile)

    # if user passed the course and currently not authorization for that run then give
    # her authorizations.
    ok_for_authorization = (
        mmtrack.has_passed_course(course_run.edx_course_key) and
        not ExamAuthorization.objects.filter(
            user=mmtrack.user,
            course=course_run.course,
            date_first_eligible__lte=now,
            date_last_eligible__gte=now
        ).exists()
    )

    if not ok_for_authorization:
        raise ExamAuthorizationException(
            message_not_passed_or_exist_template.format(
                user=mmtrack.user.username,
                course_id=course_run.edx_course_key
            )
        )

    ExamAuthorization.objects.create(
        user=mmtrack.user,
        course=course_run.course,
        date_first_eligible=now,
        date_last_eligible=add_year(now)
    )
    log.info(
        '[Exam authorization] user "%s" is authorize for exam the for course id "%s"',
        mmtrack.user.username,
        course_run.edx_course_key
    )
```